server/the-usher-openapi-spec.yaml

```diff
@@ -534,22 +536,84 @@ paths:
         500:
           $ref: '#/components/responses/InternalError'
 
+  /clients:
+    post:
+      'x-swagger-router-controller': 'clients/index'
+      operationId: createClient
+      summary: Create a new Client
+      tags:
+        - Client Admin APIs
+      security:
+        - bearerAdminAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                tenant_name:
+                  type: string
+                client_id:
+                  type: string
+                  maxLength: 50
+                  pattern: ^[a-z0-9-_]*$
+                name:
+                  type: string
+                  maxLength: 50
+                description:
+                  type: string
+                secret:
+                  type: string
+              required:
+                - tenant_name
+                - client_id
+                - name
+      responses:
+        201:
+          description: The newly created Client object
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Client'
+        400:
+          $ref: '#/components/responses/BadRequest'
+
   /clients/{client_id}:
+    parameters:
+      - $ref: '#/components/parameters/clientIdPathParam'
     get:
+      'x-swagger-router-controller': 'clients/index'
+      operationId: getClient
       summary: Get a single Client
       tags:
         - Client Admin APIs
       security:
+        - bearerAdminAuth: []
         - bearerClientAdminAuth: []
-      parameters:
-        - $ref: '#/components/parameters/clientIdPathParam'
       responses:
         200:
           description: Return details for the given Client object
           content:
             application/json:
               schema:
-                type: object
+                $ref: '#/components/schemas/Client'
+        404:
+          $ref: '#/components/responses/NotFound'
+    delete:
+      'x-swagger-router-controller': 'clients/index'
+      operationId: deleteClient
+      summary: Delete a Client and associated data
+      description: In addition to deleting the Client, any roles belonging to this Client
```

Comment:
Does it delete the associated permissions too?

Reply:
@k1ch  I just looked at the db schema and the `clientkey` column in the permissions table is not actually a foreign key to clients, therefore there is no on cascade setting currently. This should probably be a separate enhancement to make.